packages/lesswrong/components/common/CookieBanner/CookieTable.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { Components, registerComponent } from "../../../lib/vulcan-lib";
+import { CookieSignature, CookieType, CookiesTable } from "../../../lib/cookies/utils";
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {},
+  heading: {
+    padding: 0,
+    marginBottom: 8,
+    marginTop: 8,
+    fontWeight: 600,
+    fontSize: "1.2rem",
+  },
+  tableContainer: {
+    borderTopLeftRadius: theme.borderRadius.default,
+    borderTopRightRadius: theme.borderRadius.default,
+    overflow: 'hidden',
+    marginBottom: 8,
+  },
+  table: {
+    ...theme.typography.commentStyle,
+    width: "100%",
+    borderCollapse: "collapse",
+    fontWeight: 400,
+  },
+  th: {
+    backgroundColor: theme.palette.primary.main,
+    color: theme.palette.primary.contrastText,
+    padding: 8,
+    borderBottom: `1px solid ${theme.palette.grey[800]}`,
+    borderRight: `1px solid ${theme.palette.primary.main}`, // hack to make borders line up
+    borderLeft: `1px solid ${theme.palette.primary.main}`, // hack to make borders line up
+    fontSize: "1rem",
+    textAlign: "left",
+  },
+  tr: {
+    border: `1px solid ${theme.palette.grey[300]}`,
+  },
+  td: {
+    padding: 8,
+    fontSize: "1rem",
+  },
+  tdName: {
+    width: "40%",
+    wordBreak: "break-all",
+  },
+  tdDescription: {
+    width: "60%",
+  },
+});
+
+const CookieTable = ({
```

Comment:
Is this table meant to show **all** the cookies in our codebase? Because it currently doesn't do that. I'm not sure the best way to get all the cookies properly registered, maybe you just need to put them all in `cookies.ts`?

Reply:
Good catch! I've fixed this by moving them all to cookies.ts, it's a shame this is required though

I've also changed all the ones we set to show/hide stuff to be "necessary", because they aren't really relevant as far as privacy is concerned

Reply:
We also need to make sure that any cookies added in the future get added in there right? Maybe add a comment to `registerCookie()` about that.

I'm kind of confused why you moved a bunch to "necessary" though, when it feels like they fit better under "functional"? "Functional cookies: these cookies enable functionalities that are not strictly necessary for the website to be usable. This includes remembering whether certain banners are hidden, or allowing you to contact us for support via Intercom."

Reply:
I've added a comment on `registerCookie()` 👍 

I changed most of ours to "necessary" because from reading around it seems like the relevant thing as far as GDPR is concerned is whether cookies can be used to track or identify you (e.g. not doing this is part of the definition of strictly necessary cookies [here](https://www.cookieyes.com/blog/cookie-consent-exemption-for-strictly-necessary-cookies/)). But then I read the official [GDPR website](https://gdpr.eu/cookies/) and it explicitly says functional cookies include remembering your preferences, so I've changed them back to play it safe